Add new commands for android: install, run and uninstall (#429)
dotnet/xharness: README.md

You can get a specific version from [the dotnet-eng feed](https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet-eng&view=versions&package=Microsoft.DotNet.XHarness.CLI&protocolType=NuGet) where it is published.
So far we are in preview so omitting the version will fail to locate a stable version of the tool and it has to be supplied.

> To run the tool, use the `xharness` command. The tool always expects the platform (`android`/`ios`) as the first argument. To get up-to-date set of commands run `help`.

Reviewer: There was a change in master already changing this line (see where it says apple/browser) and I think you might have resolved a conflict wrong.

Reviewer: ```suggestion
To run the tool, use the `xharness` command. The tool always expects the platform (`android`/`apple`/`browser`) followed by a command. To get an up-to-date set of commands, please run `xharness help`.
```

Example for Android apk:

> You should provide path to adb before running via ADB_EXE_PATH.

Author: I need to put it when I run XHarness in VS. If it is not a generic recommendation, I may remove it

> You should provide path to adb before running via ADB_EXE_PATH.

Reviewer: Yeah, when the final customer has the .NET tool installed, it has the ADB bundled in a folder where it expects it and has a relative path pre-set pointing to this:
https://github.com/dotnet/xharness/blob/63a4360c9f9b655dcdc523ed485b63f5ecdb5e42/src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs#L71-L87

When we do F5 in Visual Studio, it needs to know, so I guess this is only needed for development. We could probably use a section in the README how to setup dev env but so far we don't have it and I think it's fine. So I wouldn't worry about this. We can deal with this later.
